src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -113,6 +114,17 @@ private async Task ProcessOutgoingMessagesAsync(ServiceConnectionContext connect
                     var buffer = result.Buffer;
                     if (!buffer.IsEmpty)
                     {
+                        // We assume the first response message would be a HandshakeResponse.
+                        if (connection.IsMigrated)
+                        {
+                            if (SignalRProtocol.HandshakeProtocol.TryParseResponseMessage(ref buffer, out var message))
```

Comment:
```cs
var handshakeBuffer = buffer;
if (SignalRProtocol.HandshakeProtocol.TryParseResponseMessage(ref handshakeBuffer, out var message))
{
  connection.IsMigrated = false;
  buffer = buffer.Slice(handshakeBuffer.Length);
  if (buffer.IsEmpty)
  {
    connection.Application.Input.AdvanceTo(buffer.End);
    continue;
  }
}
```

Reply:
No, not all `HandshakeResponse` should be intercepted.

We only care about the connection that has been migrated from another server.